index.bs

```diff
@@ -1066,6 +1066,43 @@ If the attribute identifies a URL that is auxilliary to the element's purpose, l
 
 Note: remember that attributes containing URLs should be represented in IDL as {{USVString}}; see [[#idl-string-types]].
 
+<h3 id="html-overloading">Give each HTML element a single purpose</h3>
+
+Each HTML element should have a clear purpose.
+Using HTML attributes can modify the semantics of an element,
+but attributes should not fundamentally alter that purpose.
+
+Rather than defining elements that can operate in different modes,
+having a separate element for each mode is preferable.
+This simplifies usage for authors.
+
+<div class=example>
+The <{input}> element uses a <{input/type}> attribute to switch between
+very different modes.
+Though each of these modes share a common goal of taking user input,
+that is too broad a purpose.
+The semantics of these modes could have been more clearly expressed
+through the use of distinct element names.
+</div>
+
+Overloading of semantics on the same element
+through the use of attributes could be used to select
+between different behavior.
+This only makes sense where the purpose remains the same
+for any value of the attributes.
+
+<div class=example>
+The <{textarea}> element and an <{input}> element with a <{input/type}> of `"text"`
+could have used the same element as the differences between these is not material.
+</div>
+
+Potential counterexamples include overspecialization and elements that rely on context to establish their semantics.
+The now-deprecated <{acronym}> element is duplicative of the <{abbr}> element
+and therefore overspecialization.
+The <{source}> element can have distinct purposes depending on context.
```

Comment:
This should also be in a (different) `<div class=example>`.

I disagree that this is an antipattern. The purpose of `<source>` is clear, the details change based on its context which is fine — having distinct elements would have been an antipattern just like `<caption>`/`<figcaption>`/`<legend>` that @hober mentioned.

Reply:
I think Martin agrees with you. This suggests this needs to be reworded so that people don't misread him.